lib/capistrano/s3/publisher.rb

```diff
@@ -78,8 +78,8 @@ def self.base_file_path(root, file)
           file.gsub(root, "")
         end
 
-        def self.files(deployment_path)
-          Dir.glob("#{deployment_path}/**/*")
+        def self.files(deployment_path, exclusions)
+          Dir.glob("#{deployment_path}/**/*") - Dir.glob(exclusions.map{|e| "#{deployment_path}/#{e}"})
```

Comment:
I would use spaces around } { for readability

`exclusions.map { |e| "#{deployment_path}/#{e}" }`

https://github.com/bbatsov/ruby-style-guide#spaces-braces